src/ucp/api/ucp.h

```diff
@@ -470,6 +471,14 @@ typedef struct ucp_params {
      * then this context needs thread safety.
      */
     int                                mt_workers_shared;
+
+    /**
+     * An optimization hint of how many endpoints would be created on this context.
+     * Does not affect semantics, but only transport selection criteria and the
+     * resulting performance.
```

Comment:
I would elaborate a bit. You can mention that the number can be extracted from mpirun -np ?